tests/integration/test_snap.py

```diff
@@ -138,6 +138,28 @@ def test_snap_logs():
     assert len(kp.logs(num_lines=15).splitlines()) >= 4
 
 
+def test_snap_hold_refresh():
+    cache = snap.SnapCache()
+    kp = cache["kube-proxy"]
+    kp.ensure(snap.SnapState.Latest, classic=True, channel="latest/stable")
+
+    kp.hold(duration="24h")
+
+    result = check_output(["snap", "list"])
+    assert "held" in result.decode()
```

Comment:
I'm not sure why, but the integration tests are failing right now.

That said, I don't think we should use `snap list` unless we're checking the `held` text is on the correct line. Probably better to use `snap info kube-proxy` and `'hold:' in result` so we're only looking at that one snap. Same for `test_snap_unhold_refresh` below.